src/v/cluster/types.cc

```diff
@@ -225,6 +225,19 @@ topic_table_delta::topic_table_delta(
   , previous_replica_set(std::move(previous))
   , replica_revisions(std::move(replica_revisions)) {}
 
+model::revision_id
+topic_table_delta::get_replica_revision(model::node_id replica) const {
+    vassert(
+      replica_revisions, "ntp {}: replica_revisions map must be present", ntp);
+    auto rev_it = replica_revisions->find(replica);
+    vassert(
+      rev_it != replica_revisions->end(),
+      "ntp {}: node {} must be present in the replica_revisions map",
+      ntp,
+      replica);
```

Comment:
nit:since this is public api of `topic_table_delta` we can not assume that it always will be asked with a correct replica, maybe it would be better to return an optional and verify at call site ?

Reply:
The point of this function was to avoid writing these checks at the call site :) I'll add a comment about pre-conditions.